htlcswitch/switch.go

```diff
@@ -492,13 +487,12 @@ func (s *Switch) forward(packet *htlcPacket) error {
 			} else {
 				failure = lnwire.NewTemporaryChannelFailure(update)
 			}
-			addErr := ErrIncompleteForward
 
 			linkError := NewDetailedLinkError(
 				failure, OutgoingFailureIncompleteForward,
 			)
 
-			return s.failAddPacket(packet, linkError, addErr)
+			return s.failAddPacket(packet, linkError)
```

Comment:
I would squash this commit with the previous, to avoid touching the same lines twice in a PR. `linkError` replaces `addErr` and it is nice to see that in one diff, but this is personal preference.